knowledge_graphs/ai_script_analyzer.py

```
@@ -0,0 +1,330 @@
+                class_info = {
+                    "name": node.name,
+                    "bases": [self._get_name_from_node(base) for base in node.bases],
+                    "line_number": node.lineno,
+                    "has_docstring": bool(ast.get_docstring(node)),
+                    "methods": [],
+                    "decorators": [self._get_decorator_name(dec) for dec in node.decorator_list]
+                }
+                
+                # Extract methods
+                for item in node.body:
+                    if isinstance(item, (ast.FunctionDef, ast.AsyncFunctionDef)):
```

Ruff (0.12.2)

Use `X | Y` in `isinstance` call instead of `(X, Y)`

Convert to `X | Y`

(UP038)